sql/schemachanger: Correct version gate for ALTER COLUMN TYPE

spilchen opened the pull request:
Previously, a version gate of 24.2 was applied to certain ALTER COLUMN TYPE operations. For trivial or validation-only changes, support for ALTER COLUMN TYPE in the declarative schema changer was added in PRs
 #126143 and #127823. However, the wrong version gate was used; these
changes were first introduced in version 24.3.0. This update corrects the version gate to reflect that.

Epic: CRDB-25314
Closes #132523
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/135731)
<!-- Reviewable:end -->

spilchen committed:
sql/schemachanger: Correct version gate for ALTER COLUMN TYPE

Previously, a version gate of 24.2 was applied to certain ALTER COLUMN
TYPE operations. For trivial or validation-only changes, support for
ALTER COLUMN TYPE in the declarative schema changer was added in PRs
 #126143 and #127823. However, the wrong version gate was used; these
changes were first introduced in version 24.3.0. This update corrects
the version gate to reflect that.

Epic: CRDB-25314
Closes #132523
Release note: none

fqazi reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135731#-:-OC4oAWlCApOy53dF96q:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @spilchen)



<!-- Sent from Reviewable.io -->

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135731#-:-OC4oGiF-d3D66cQBE2T:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @spilchen)



<!-- Sent from Reviewable.io -->

spilchen commented:
TFTR! 

bors r+
